src/libAtomVM/bitstring.c

```diff
@@ -158,6 +224,55 @@ bool bitstring_utf16_encode(avm_int_t c, uint8_t *buf, enum BitstringFlags bs_fl
     return true;
 }
 
+bool bitstring_utf16_decode(int32_t *c, uint8_t *buf, size_t len, enum BitstringFlags bs_flags, size_t *out_size)
+{
+    *c = 0;
+    if (len == 0) {
+        return false;
+    } else if (bs_flags & LittleEndianIntegerMask) {
+        if (len >= 4 && ((*(buf + 1) & 0xD8) == 0xD8) && ((*(buf + 3) & 0xDC) == 0xDC)) {
+            *c |= (*(buf + 1) & 0x03) << 18;
+            *c |= (*(buf + 0) & 0xFF) << 10;
+            *c |= (*(buf + 3) & 0x03) << 8;
+            *c |= (*(buf + 2) & 0xFF);
+            *c += 0x10000;
+            if (*c < 0 || *c > 0x10FFFF) {
```

Comment:
This cannot happen. UTF-16 can only encode between 0 & 16#FFFF (and when we have surrogates, the values are from 16#010000 and 16#10FFFF.